### PR TITLE
add an improved color scheme for the flamegraph

### PR DESCRIPTION
**Pull request by lievenhey**
the new color scheme selects the color based on the cost in % an very expensive function is colored red, a cheap function is colored yellow

### CODE REVIEW

**Comment by lievenhey** [comment]
grafik

File: src/flamegraph.cpp [hunk 1]
```diff
@@ -365,7 +365,16 @@ QBrush brushSystem(const Data::Symbol& symbol)
     return user;
 }
 
-QBrush brush(const Data::Symbol& entry, Settings::ColorScheme scheme)
+QBrush improvedBrush(quint32 cost, quint32 totalCost)
```
Comment (milianw) [inline src/flamegraph.cpp:368]: "improved" is very subjective. I don't think it improves anything. please give it a different name, something like `brushByCostRatio` or similar

File: src/flamegraph.cpp [hunk 2]
```diff
@@ -713,6 +724,7 @@ FlameGraph::FlameGraph(QWidget* parent, Qt::WindowFlags flags)
     m_colorSchemeSelector->addItem(tr("Binary"), QVariant::fromValue(Settings::ColorScheme::Binary));
     m_colorSchemeSelector->addItem(tr("Kernel"), QVariant::fromValue(Settings::ColorScheme::Kernel));
     m_colorSchemeSelector->addItem(tr("System"), QVariant::fromValue(Settings::ColorScheme::System));
+    m_colorSchemeSelector->addItem(tr("Improved"), QVariant::fromValue(Settings::ColorScheme::Improved));
```
Comment (milianw) [inline src/flamegraph.cpp:727]: Cost Ratio

File: src/settings.h [hunk 3]
```diff
@@ -23,6 +23,7 @@ class Settings : public QObject
         Binary,
         Kernel,
         System,
+        Improved,
```
Comment (milianw) [inline src/settings.h:26]: CostRatio

**Review by milianw — CHANGES_REQUESTED** [review]
I'm fine with adding it, but not under this name. it doesn't improve anything, it's just a different way to colorize it - and note how it essentially duplicates information that we already represent in the flamegraph: the width already encodes the cost ratio, so doing it again with the color... what for?

**Comment by milianw** [comment]
please rebase, the compile errors should be gone

**Comment by lievenhey** [comment]
done

**Review by milianw — APPROVED** [review]

**Pull request merged**